susemanager-utils/mgr-libmod/mgrlibmod/mllib.py

```diff
@@ -0,0 +1,330 @@
+        assert self._mod_index is not None, "Unable to get module streams: module index not found"
+
+        streams: Set = set()
+        module = self._mod_index.get_module(name)
+        if module:
+            for s_obj in module.get_all_streams():
+                streams.add(s_obj.get_stream_name())
+
+        return list(streams)
+
+    def get_stream_contexts(self, s_type: mltypes.MLStreamType) -> List:
+        if self._mod_index is None:
+            self.index_modules()
+        assert self._mod_index is not None, "Unable to get stream contexts: module index not found"
+        contexts: List = []
+        module = self._mod_index.get_module(s_type.name)
+        if module:
+            for stream in module.get_all_streams():
+                if stream.get_stream_name() == s_type.stream:
+                    contexts.append(stream)
+
+        return contexts
+
+    def get_stream_dependencies(self, ctx: Modulemd.ModuleStreamV2) -> List[str]:
+        """
+        get_stream_dependencies -- get stream dependencies.
+
+        :param ctx: module stream context
+        :type ctx: Modulemd.ModuleStreamV2
+        :return: list of dependency names.
+        :rtype: List[str]
+        """
+        deps: List[str] = []
+        s_deps: List[Modulemd.Dependencies] = ctx.get_dependencies() or []
+        dep: Modulemd.Dependencies
+        for dep in s_deps:
+            deps.extend(dep.get_runtime_modules())
+
+        return deps
+
+    def index_modules(self) -> None:
+        """
+        index_modules -- loads given metadata and indexes modules from there.
+        """
+        if self._mod_index is None:
+            mgr: Modulemd.ModuleIndex = Modulemd.ModuleIndexMerger.new()
+            for path in self.metadata:
+                idx = Modulemd.ModuleIndex.new()
+                if self._is_meta_compressed(path):
+                    with gzip.open(path) as gzmeta:
+                        idx.update_from_string(gzmeta.read().decode("utf-8"), True)
+                else:
+                    idx.update_from_file(path, True)
+                mgr.associate_index(idx, 0)
+            self._mod_index = mgr.resolve()
+
+    def get_default_stream(self, name: str):
+        assert self._mod_index is not None, "Unable to access module index when resolving default stream"
+        module = self._mod_index.get_module(name)
+        if not module:
+            raise ValueError("Module {} not found".format(name))
+        defaults = module.get_detaults()
+        if defaults:
+            return defaults.get_default_stream()
+
+        return module.get_all_streams()[0].get_stream_name()
+
+    def get_dep_streams(self, s_obj):
+        dep = s_obj.get_dependencies()[0]  # XXX: Why just always first?
+        all_deps = []  # type: ignore
+        for m in dep.get_runtime_modules():
+            deps = dep.get_runtime_streams(m)
+            if deps:
+                all_deps.append((m, deps[0],))  # XXX: why just first?
```

Comment:
we're calling `get_runtime_streams(m)` per module, and by definition, there cannot be more than 1 stream enabled for a module at the same time. Therefore, it's not possible for a module to depend on multiple streams of a same module. So in the end, this should also always be a 1 element array.

But again, I'm not against foolproofing this either.

Reply:
Yeah, let's foolproof. Usually the `foo[idx]` code in Python world considered evil, unless you _really_ know what you are doing. E.g. in unit tests it is OK, but in code like that, _maybe_ OK. I am still wondering why they are returning array with one element than just element...

Reply:
I would resolve this for now.